File: code/bones/java-kotlin-scala/cloud/jwt/jwt-boot/src/main/java/com/sf/jwt/domain/service/UserServiceImpl.java
package com.sf.jwt.domain.service;

import com.sf.jwt.domain.entity.User;
import com.sf.jwt.domain.repository.mapper.UserMapper;
import org.apache.commons.lang3.StringUtils;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Repository;
import org.springframework.stereotype.Service;

@Service
public class UserServiceImpl {
   @Autowired
    UserMapper userMapper;

    public User getUser(String name,String password){
        if(StringUtils.isBlank(name) || StringUtils.isBlank(password)){
            return null;
        }
        return userMapper.selectByNameAndPassword(name,password);
    }
}

File: code/bones/java-kotlin-scala/jktdemo/src/main/java/com/sf/jkt/k/auto/service/impl/UserServiceImpl.kt
package com.sf.jkt.k.auto.service.impl;

import com.sf.jkt.k.auto.entity.User;
import com.sf.jkt.k.auto.mapper.UserMapper;
import com.sf.jkt.k.auto.service.IUserService;
import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl;
import org.springframework.stereotype.Service;

/**
 * <p>
 *  服务实现类
 * </p>
 *
 * @author songfei
 * @since 2019-08-14
 */
@Service
open class UserServiceImpl : ServiceImpl<UserMapper, User>(), IUserService {

}
